### PR TITLE
Add LDAPIdentityProvider.spec.groupSearch.userAttributeForFilter

### PR DESCRIPTION
**Pull request by cfryanr**
Addresses #1533.

**Release note**:

```release-note
Added new `LDAPIdentityProvider.spec.groupSearch.userAttributeForFilter` and 
`ActiveDirectoryIdentityProvider.spec.groupSearch.userAttributeForFilter` configuration options.
The additional flexibility for LDAP and AD group searches introduced by this new configuration option can
be used to find groups in new ways, such as finding groups defined using the `posixGroup` schema.
For backwards compatibility, the group search defaults to the old behavior when this new option is not set.
For more details, see the API documentation https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.27/README.adoc#ldapidentityprovidergroupsearch.
```

### CODE REVIEW

**Comment by cfryanr** [comment]
I manually tested the code committed so far on this WIP PR and it works.

Next, need to write some integration tests, which:
1. Needs to work when run locally after using `prepare-for-integration-tests.sh`.
2. Needs to work in CI on Kind clusters with the openldap server running in the same cluster.
3. Needs to work in CI when run against JumpCloud's LDAP, or needs to be skipped in that case.

**Comment by cfryanr** [comment]
Integration tests added!

File: test/integration/ldap_client_test.go [hunk 1]
```diff
@@ -347,6 +347,98 @@ func TestLDAPSearch_Parallel(t *testing.T) {
...
+				ExtraRefreshAttributes: map[string]string{},
+			},
+		},
+		{
+			name:     "using a group search with UserAttributeForFilter set to sn",
+			username: "pinny",
+			password: pinnyPassword,
+			provider: upstreamldap.New(*providerConfig(func(p *upstreamldap.ProviderConfig) {
+				p.GroupSearch.Filter = "&(objectClass=posixGroup)(memberUid={})"
+				p.GroupSearch.UserAttributeForFilter = "givenName" // pinny's givenName is not "pinny" so it should not find any groups, and also should not error on the emoji in the givenName
```
Comment (joshuatcasey) [inline test/integration/ldap_client_test.go:398]: givenName or sn? The test name and the userAttributeForFilter are not in sync

**Review by joshuatcasey — APPROVED** [review]

File: test/integration/ldap_client_test.go (hunk 1 above)
Comment (cfryanr) [reply test/integration/ldap_client_test.go:398]: Fixed and force pushed the fix.

**Pull request merged**